lib/pure/typetraits.nim

```diff
@@ -263,3 +263,14 @@ since (1, 1):
 
     type T2 = T
     genericParamsImpl(T2)
+
+
+proc hasClosureImpl(n: NimNode): bool = discard "see compiler/vmops.nim"
+
+proc hasClosure*(fn: NimNode): bool {.since: (1, 5, 1).} =
+  ## Return true if the func/proc/etc `fn` has `closure`.
```

Comment:
This doc comment should be improved in the future, I for one have no idea what "if [...] `fn` has `closure`" is supposed to mean. That `fn` is a closure? Or is `closure` some field or flag?

Also, an example would be nice.